commit e788cf7380fb3ff18fed99ca2e3fa968d0334d84
Author: Ravindu Deshan <droidcasts@protonmail.com>
Date:   Thu Mar 27 00:35:02 2025 +0530

    kernel: Use the stock config for /proc/config.gz
    
    Userspace reads /proc/config.gz and spits out an error message after boot
    finishes when it doesn't like the kernel's configuration. In order to
    preserve our freedom to customize the kernel however we'd like, show
    userspace the stock defconfig so that it never complains about our
    kernel configuration.
    
    Signed-off-by: Ravindu Deshan <droidcasts@protonmail.com>

diff --git a/kernel/Makefile b/kernel/Makefile
index 3d6ca8e9d..3fc4dc811 100644
--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -120,7 +120,7 @@ obj-$(CONFIG_HAS_IOMEM) += memremap.o
 $(obj)/configs.o: $(obj)/config_data.h
 
 targets += config_data.gz
-$(obj)/config_data.gz: $(KCONFIG_CONFIG) FORCE
+$(obj)/config_data.gz: arch/arm64/configs/stock_defconfig FORCE
 	$(call if_changed,gzip)
 
       filechk_ikconfiggz = (echo "static const char kernel_config_data[] __used = MAGIC_START"; cat $< | scripts/basic/bin2c; echo "MAGIC_END;")
